dashboard/frontend/src/components/CreateJob.js

```diff
@@ -0,0 +1,201 @@
+          <TextField floatingLabelText="Container image" name="workerImage" value={this.state.workerImage} onChange={this.handleInputChange} />
+          <TextField floatingLabelText="Replicas" type="number" min="0" name="workerReplicas" value={this.state.workerReplicas} onChange={this.handleInputChange} />
+          <TextField floatingLabelText="GPU(s) per replica" type="number" min="0" name="workerGpuCount" value={this.state.workerGpuCount} onChange={this.handleInputChange} />
+
+          {/* PARAMETER SERVER */}
+          <Divider style={this.styles.divider} />
+          <p style={this.styles.header}>Parameter Server(s)</p>
+          <TextField floatingLabelText="Replicas" name="psReplicas" type="number" min="0" value={this.state.psReplicas} onChange={this.handleInputChange} />
+          <Toggle label="Use default image" defaultToggled={true} name="psUseDefaultImage" onToggle={this.handleInputChange} style={this.styles.toggle} />
+          {!this.state.psUseDefaultImage &&
+            <TextField floatingLabelText="Container image" name="psImage" value={this.state.psImage} onChange={this.handleInputChange} />
+          }
+
+          {/* TENSORBOARD */}
+          <Divider style={this.styles.divider} />
+          <Toggle label="TensorBoard" defaultToggled={true} name="tbIsPresent" onToggle={this.handleInputChange} style={this.styles.toggle} />
+          {this.state.tbIsPresent &&
+            <CreateTensorBoard setTensorBoardSpec={this.setTensorboardSpec} />
+          }
+        </CardText>
+        <CardActions>
+          <RaisedButton label="Deploy" primary={true} onClick={this.deploy} />
+          <RaisedButton label="Cancel" onClick={this.cancel} />
+        </CardActions>
+      </Card >
+    );
+  } 
+
+  deploy() {
+    console.log(this.state.tbSpec);
+  }
+
+  deploy2() {
+
+    let rs = [
+      this.newReplicaSpec("MASTER", 1, this.state.masterImage)
+    ]
+    if (this.state.workerReplicas > 0) {
+      rs.push(this.newReplicaSpec("WORKER", this.state.workerReplicas, this.state.workerImage));
+    }
+    if (this.state.psReplicas > 0) {
+      rs.push(this.newReplicaSpec("PS", this.state.psReplicas, this.state.psImage));
+    }
+
+    let spec = {
+      metadata: {
+        name: this.state.name,
+        namespace: this.state.namespace
+      },
+      spec: {
+        replicaSpecs: rs,
+
+      }
+    }
+
+    if (this.state.tbIsPresent) {
+      spec.spec.tensorboard = this.tensorBoard.getTensorBoardSpec();
+    }
+
+    createTfJobService(spec)
+      .catch(console.log);
+  }
+
+  cancel() {
+    this.props.history.goBack();
+  }
+
+  newReplicaSpec(tfReplicaType, replicas, image) {
+    return {
+      replicas: parseInt(replicas),
+      tfReplicaType,
+      template: {
+        spec: {
+          containers: [{
+            image,
+            name: "tensorflow"
+          }],
+          restartPolicy: "OnFailure"
+        }
+      }
+
+    }
+  }
+
+  buildTensorBoardSpec() {
+    // "tensorboard": {
```

Comment:
Can we delete this commented out code?

Reply:
This needs to be finished before merging. Missing volumes and volumemount.